db/models/common_functions.py

```diff
@@ -0,0 +1,180 @@
+            ):
+                form.status = "COMPLETED"
+                continue
+            elif (
+                question["status"] == "NOT_STARTED"
+                and form.status == "COMPLETED"
+            ):
+                form.status = "IN_PROGRESS"
+                continue
+            elif question["status"] == "IN_PROGRESS":
+                form.status = "IN_PROGRESS"
+                break
+    db.session.commit()
+
+
+def update_question_statuses(application_id: str):
+    """
+    Updates the question statuses of each question if a value is present
+
+    Args:
+        application_id: The application id
+    """
+
+    forms = FormsMethods.get_forms_by_app_id(application_id, as_json=False)
+    for form in forms:
+        for question in form.json:
+
+            try:
+
+                question_status = question["status"]
+
+            except KeyError:
+
+                question["status"] = "NOT_STARTED"
+
+                question_status = question["status"]
+
+            if question_status == "SUBMITTED":
+
+                break
+
+            answer_found_list = [
+                field["answer"] not in [None, ""]
+                for field in question["fields"]
+            ]
+
+            # If all answers are given
+            if all(answer_found_list):
+                question["status"] = "COMPLETED"
+            # If no answers are given
+            elif not all(
+                [not found_answer for found_answer in answer_found_list]
+            ):
+                question["status"] = "IN_PROGRESS"
+            # If some answers are given
+            else:
+                question["status"] = "NOT_STARTED"
+
+    db.session.commit()
+
+
+def update_statuses(application_id):
+
+    update_question_statuses(application_id)
+    update_form_statuses(application_id)
+    update_application_status(application_id)
+
+
+def get_application_bundle_by_id(app_id):
+
+    application = ApplicationsMethods.get_application_by_id(app_id)
+
+    forms = FormsMethods.get_forms_by_app_id(app_id)
+
+    return {**application.as_dict(), "forms": forms}
+
+
+def submit_application(application_id):
+
+    application = ApplicationsMethods.get_application_by_id(application_id)
+
+    application.date_submitted = datetime.datetime.now(
+        datetime.timezone.utc
+    ).strftime("%Y-%m-%d %H:%M:%S")
+
+    db.commit()
+
+    update_statuses(application_id)
+
+    account = AccountMethods.get_account(
+        account_id=application.get("account_id")
+    )
+
+    # Send notification
+    Notification.send(
+        Config.NOTIFY_TEMPLATE_SUBMIT_APPLICATION,
+        account.email,
+        # TODO
+        {
+            "application": ApplicationsMethods.get_application_bundle_by_id(
+                application_id
+            )
+        },
+    )
+
+    return ApplicationsMethods.get_application_bundle_by_id(application_id)
+
+
+def update_form(application_id, form_name, question_json):
```

Comment:
And this could maybe go in to FormMethods?

Reply:
from memory i think we put these in a common file to avoid circular import issues - correct me if this is wrong @SentientHousePlant